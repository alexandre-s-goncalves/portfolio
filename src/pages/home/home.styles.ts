import styled from 'styled-components';
import {Button} from 'components/ui/button';
import {Text} from 'components/ui/text';
import {Icon} from 'components/icon';
import {colors} from 'resources/colors';
import {margin} from 'resources/margins';
import ArrowSVG from 'assets/icons/iArrow.svg';

interface ThemeProps {
  $themeDark?: boolean;
}

export const ContactButton = styled(Button).attrs<ThemeProps>(({$themeDark}) => ({
  $hoverBackgroundColor: colors.background03,
  textProps: {
    color: $themeDark ? colors.white : colors.darkContent,
  },
}))`
  background-color: transparent;
  border: 1px solid ${colors.neutralLight};
`;

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100vw;
`;

export const ContentWrapper = styled.div`
  display: flex;
  align-items: center;
  gap: ${margin.small}px;
  margin-bottom: ${margin.large}px;
`;

export const Description = styled(Text).attrs({
  size: 18,
  lineHeight: 22,
  color: colors.gray,
})`
  margin: ${margin.xlarge}px 0;
  width: 700px;
  text-align: center;
`;

export const EmailIcon = styled(Icon).attrs<ThemeProps>(({$themeDark}) => ({
  width: 32,
  height: 32,
  color: $themeDark ? colors.grayLight : colors.black,
}))``;

export const GithubIcon = styled(Icon).attrs<ThemeProps>(({$themeDark}) => ({
  width: 24,
  height: 24,
  color: $themeDark ? colors.grayLight : colors.black,
}))<ThemeProps>``;

export const Greeting = styled(Text).attrs({
  size: 18,
  lineHeight: 26,
  color: colors.gray,
})`
  margin-bottom: ${margin.small}px;
`;

export const LinkedinIcon = styled(Icon).attrs({
  width: 24,
  height: 24,
})``;

export const Name = styled(Text).attrs<ThemeProps>(({$themeDark}) => ({
  type: 'h1',
  size: 40,
  lineHeight: 40,
  weight: 'bold',
  color: $themeDark ? colors.white : colors.darkContent,
}))`
  margin-bottom: ${margin.small}px;
`;

export const Role = styled(Text).attrs({
  size: 26,
  lineHeight: 34,
  color: colors.gray,
})``;

export const SocialLink = styled.a`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  width: 40px;
  height: 40px;

  svg {
    transition: transform 0.2s ease;
  }

  &:hover svg {
    transform: scale(1.1);
  }
`;

export const SocialLinks = styled.div`
  display: flex;
  align-items: center;
  gap: ${margin.small}px;
`;

export const Tag = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 8px;
  padding: 6px 12px;
  background-color: ${colors.neutralXLight};
`;

export const TagText = styled(Text).attrs({
  size: 12,
  lineHeight: 16,
  weight: 'semi-bold',
  color: colors.neutralDark,
})``;

export const ViewProjectsButton = styled(Button).attrs<ThemeProps>(({$themeDark}) => ({
  icon: ArrowSVG,
  iconPosition: 'right',
  textProps: {
    color: $themeDark ? colors.darkContent : colors.white,
  },
}))<ThemeProps>`
  background-color: ${({$themeDark}) => ($themeDark ? colors.white : colors.darkContent)};
`;
