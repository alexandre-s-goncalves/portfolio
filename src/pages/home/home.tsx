import React from 'react';
import {useTranslation} from 'react-i18next';
import {useTheme} from 'context/ThemeContext';
import {namespaces} from 'utils/i18n/i18n.constants';
import EmailSVG from 'assets/icons/iEmail.svg';
import GithubSVG from 'assets/icons/iGithub.svg';
import LinkedinSVG from 'assets/icons/iLinkedin.svg';
import * as S from './home.styles';

export const Home = () => {
  const {t: tFooter} = useTranslation(namespaces.pages.footer);
  const {t} = useTranslation(namespaces.pages.home);
  const {themeDark} = useTheme();
  const tags = ['React.js', 'TypeScript', 'React Native', '.NET', 'Java'];

  return (
    <S.Container data-testid="home-page">
      <S.Greeting>{t('greeting')}</S.Greeting>
      <S.Name $themeDark={themeDark}>{t('name')}</S.Name>
      <S.Role>{t('role')}</S.Role>
      <S.Description>{t('description')}</S.Description>
      <S.ContentWrapper>
        {tags.map(tag => (
          <TagItem key={tag} tag={tag} />
        ))}
      </S.ContentWrapper>
      <S.ContentWrapper>
        <S.ViewProjectsButton $themeDark={themeDark}>{t('view_projects')}</S.ViewProjectsButton>
        <S.ContactButton $themeDark={themeDark}>{t('contact')}</S.ContactButton>
      </S.ContentWrapper>
      <S.SocialLinks>
        <S.SocialLink
          href={tFooter('github')}
          target="_blank"
          rel="noopener noreferrer"
          aria-label={tFooter('github')}>
          <S.GithubIcon $themeDark={themeDark} icon={GithubSVG} />
        </S.SocialLink>
        <S.SocialLink
          href={tFooter('linkedin')}
          target="_blank"
          rel="noopener noreferrer"
          aria-label={tFooter('linkedin')}>
          <S.LinkedinIcon icon={LinkedinSVG} />
        </S.SocialLink>
        <S.SocialLink href={`mailto:${tFooter('email')}`} aria-label="Send email">
          <S.EmailIcon $themeDark={themeDark} icon={EmailSVG} />
        </S.SocialLink>
      </S.SocialLinks>
    </S.Container>
  );
};

const TagItem = ({tag}: {tag: string}) => {
  return (
    <S.Tag>
      <S.TagText>{tag}</S.TagText>
    </S.Tag>
  );
};
