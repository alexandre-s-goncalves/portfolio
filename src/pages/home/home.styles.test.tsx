import React from 'react';
import {render} from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import * as S from './home.styles';
import EmailSVG from 'assets/icons/iEmail.svg';
import GithubSVG from 'assets/icons/iGithub.svg';

describe('home.styles', () => {
  describe('Container', () => {
    test('should match the snapshot', () => {
      const {container} = render(<S.Container />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Name', () => {
    test('should match the snapshot with light theme', () => {
      const {container} = render(<S.Name $themeDark={false} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should match the snapshot with dark theme', () => {
      const {container} = render(<S.Name $themeDark={true} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('ContactButton', () => {
    test('should match the snapshot with light theme', () => {
      const {container} = render(<S.ContactButton $themeDark={false}>Contact</S.ContactButton>);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should match the snapshot with dark theme', () => {
      const {container} = render(<S.ContactButton $themeDark={true}>Contact</S.ContactButton>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('EmailIcon', () => {
    test('should render with light theme', () => {
      const {container} = render(<S.EmailIcon $themeDark={false} icon={EmailSVG} />);
      expect(container.firstChild).toBeInTheDocument();
    });

    test('should render with dark theme', () => {
      const {container} = render(<S.EmailIcon $themeDark={true} icon={EmailSVG} />);
      expect(container.firstChild).toBeInTheDocument();
    });
  });

  describe('GithubIcon', () => {
    test('should render with light theme', () => {
      const {container} = render(<S.GithubIcon $themeDark={false} icon={GithubSVG} />);
      expect(container.firstChild).toBeInTheDocument();
    });

    test('should render with dark theme', () => {
      const {container} = render(<S.GithubIcon $themeDark={true} icon={GithubSVG} />);
      expect(container.firstChild).toBeInTheDocument();
    });
  });

  describe('ViewProjectsButton', () => {
    test('should match the snapshot with light theme', () => {
      const {container} = render(
        <S.ViewProjectsButton $themeDark={false}>View Projects</S.ViewProjectsButton>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should match the snapshot with dark theme', () => {
      const {container} = render(
        <S.ViewProjectsButton $themeDark={true}>View Projects</S.ViewProjectsButton>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
