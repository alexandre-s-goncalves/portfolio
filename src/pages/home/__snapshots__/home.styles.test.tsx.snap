// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home.styles ContactButton should match the snapshot with dark theme 1`] = `
.c2 {
  font-size: 14px;
  font-family: Inter-Regular;
  color: #ffffff;
  margin-top: 0px;
  font-weight: normal;
  text-transform: none;
  cursor: text;
}

.c0 {
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 8px;
  padding: 8px 24px;
  border-radius: 8px;
  border: none;
  height: 40px;
  transition: all 0.2s ease;
  background-color: #d3d3d3;
  cursor: pointer;
}

.c0:hover {
  opacity: 0.9;
}

.c0:active {
  transform: scale(0.98);
}

.c0:disabled {
  cursor: default;
}

.c0:disabled:hover {
  opacity: 1;
}

.c3 {
  white-space: nowrap;
  cursor: inherit;
}

.c1 {
  background-color: transparent;
  border: 1px solid #D9DDDB;
}

<button
  aria-label="Contact"
  class="c0 c1"
  type="button"
>
  <p
    class="c2 c3"
  >
    Contact
  </p>
</button>
`;

exports[`home.styles ContactButton should match the snapshot with light theme 1`] = `
.c2 {
  font-size: 14px;
  font-family: Inter-Regular;
  color: #0a0a0a;
  margin-top: 0px;
  font-weight: normal;
  text-transform: none;
  cursor: text;
}

.c0 {
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 8px;
  padding: 8px 24px;
  border-radius: 8px;
  border: none;
  height: 40px;
  transition: all 0.2s ease;
  background-color: #d3d3d3;
  cursor: pointer;
}

.c0:hover {
  opacity: 0.9;
}

.c0:active {
  transform: scale(0.98);
}

.c0:disabled {
  cursor: default;
}

.c0:disabled:hover {
  opacity: 1;
}

.c3 {
  white-space: nowrap;
  cursor: inherit;
}

.c1 {
  background-color: transparent;
  border: 1px solid #D9DDDB;
}

<button
  aria-label="Contact"
  class="c0 c1"
  type="button"
>
  <p
    class="c2 c3"
  >
    Contact
  </p>
</button>
`;

exports[`home.styles Container should match the snapshot 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100vw;
}

<div
  class="c0"
/>
`;

exports[`home.styles Name should match the snapshot with dark theme 1`] = `
.c0 {
  font-size: 40px;
  font-family: Inter-Bold;
  color: #ffffff;
  margin-top: 0px;
  font-weight: normal;
  text-transform: none;
  cursor: text;
  line-height: 40px;
}

.c1 {
  margin-bottom: 16px;
}

<h1
  class="c0 c1"
/>
`;

exports[`home.styles Name should match the snapshot with light theme 1`] = `
.c0 {
  font-size: 40px;
  font-family: Inter-Bold;
  color: #0a0a0a;
  margin-top: 0px;
  font-weight: normal;
  text-transform: none;
  cursor: text;
  line-height: 40px;
}

.c1 {
  margin-bottom: 16px;
}

<h1
  class="c0 c1"
/>
`;

exports[`home.styles ViewProjectsButton should match the snapshot with dark theme 1`] = `
.c2 {
  font-size: 14px;
  font-family: Inter-Regular;
  color: #0a0a0a;
  margin-top: 0px;
  font-weight: normal;
  text-transform: none;
  cursor: text;
}

.c0 {
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 8px;
  padding: 8px 24px;
  border-radius: 8px;
  border: none;
  height: 40px;
  transition: all 0.2s ease;
  background-color: #d3d3d3;
  cursor: pointer;
}

.c0:hover {
  opacity: 0.9;
}

.c0:active {
  transform: scale(0.98);
}

.c0:disabled {
  cursor: default;
}

.c0:disabled:hover {
  opacity: 1;
}

.c3 {
  white-space: nowrap;
  cursor: inherit;
}

.c1 {
  background-color: #ffffff;
}

<button
  aria-label="View Projects"
  class="c0 c1"
  type="button"
>
  <p
    class="c2 c3"
  >
    View Projects
  </p>
  <svg
    class=""
    fill="#0a0a0a"
    height="24"
    style="min-width: 24px; min-height: 24px; max-width: 24px; max-height: 24px;"
    width="24"
  />
</button>
`;

exports[`home.styles ViewProjectsButton should match the snapshot with light theme 1`] = `
.c2 {
  font-size: 14px;
  font-family: Inter-Regular;
  color: #ffffff;
  margin-top: 0px;
  font-weight: normal;
  text-transform: none;
  cursor: text;
}

.c0 {
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 8px;
  padding: 8px 24px;
  border-radius: 8px;
  border: none;
  height: 40px;
  transition: all 0.2s ease;
  background-color: #d3d3d3;
  cursor: pointer;
}

.c0:hover {
  opacity: 0.9;
}

.c0:active {
  transform: scale(0.98);
}

.c0:disabled {
  cursor: default;
}

.c0:disabled:hover {
  opacity: 1;
}

.c3 {
  white-space: nowrap;
  cursor: inherit;
}

.c1 {
  background-color: #0a0a0a;
}

<button
  aria-label="View Projects"
  class="c0 c1"
  type="button"
>
  <p
    class="c2 c3"
  >
    View Projects
  </p>
  <svg
    class=""
    fill="#ffffff"
    height="24"
    style="min-width: 24px; min-height: 24px; max-width: 24px; max-height: 24px;"
    width="24"
  />
</button>
`;
